src/components/utils/math.js

```diff
@@ -0,0 +1,26 @@
+const linearMapping = (intendedStart, intendedEnd, Xstart, Xend, X) =>
+    (X-Xstart)/(Xend-Xstart) * (intendedEnd - intendedStart) + intendedStart
+
+const centeredPos = (sourceWidth, intendedWidth) => (sourceWidth - intendedWidth)/2
```

Comment:
These variable names aren't really accurate. The function name confuses me a bit, too.

Does `const getMarginToCenter = (outerLength, innerLength) => (outerLength - innerLength) / 2` make more sense to you?